src/MessageComparator.ts

```diff
@@ -0,0 +1,24 @@
+import { IMessage } from 'botbuilder'
+
+export class MessageComparator {
+  private message: IMessage;
+
+  constructor(message: IMessage) {
+    this.message = message;
+  }
+
+  public isEqual(msg: IMessage): boolean {
+    return (function deepInclude(message: any, otherMessage: any): boolean {
```

Comment:
I'd recommend making a private function that you call here. This is not immediately clear what is happening